src/qz/ws/PrintSocketServer.java

```diff
@@ -49,8 +50,13 @@
     public static final List<Integer> INSECURE_PORTS = Collections.unmodifiableList(Arrays.asList(8182, 8283, 8384, 8485));
 
 
-    private static TrayManager trayManager;
-    private static Properties trayProperties;
+    private static TrayManager trayManager = null;
+    private static Properties trayProperties = null;
+    private static Server server = null;
+
+    private static final AtomicBoolean running = new AtomicBoolean(true);
```

Comment:
Why has `running` changed to be true by default?
The `runServer` loop appears only loop on a `true` state for this value now too (L138), which means it will always loop through every port in the lists when ran. We want this to break on the first successful connection.

Reply:
> which means it will always loop through every port in the lists when ran

Actually, it does not, it stops after it finds one, because `server.join()` blocks. If that wasn't true the program would simply exit once `running` was set to true since the loop would stop and it would exit.

> Why has running changed to be true by default?

For reload to work the loop needs to run again when reload is called. Running refers to the overall state of the program, so as long as it's running, if the server is stopped, start it up again.